packages/query/src/index.ts

```diff
@@ -183,10 +206,10 @@ export class LiveQuery extends TxProcessor implements Storage, Queriable {
 
         if (q.options?.limit !== undefined && q.result.length > q.options.limit) {
           if (q.result.pop()?._id !== doc._id) {
-            q.callback(Object.assign(q.result, { total: q.total }))
+            q.callback(Object.assign(copy(q.result), { total: q.total }))
```

Comment:
You could use Object.assign({}, q.result, { total: q.total })

Reply:
Couldn't, it is not working...